### PR TITLE
Add details for .NET 5 TFMs 

### LINKED ISSUE #21304: .NET 5 TFM should include more in-depth explanation of platform and version parts
The .NET 5 new TFMs as described here: https://docs.microsoft.com/en-us/dotnet/standard/frameworks#net-5-os-specific-tfms don't really describe how exactly to use these. It links to the design doc, which is not a great "docs" for this feature (it explains the whys and hows and has lot of details), we should definitely keep the link for more details, but provide a high-level version in the docs site. Ideally we would also link to the platform specific docs for each, for example to this doc about the Windows TFMs: https://docs.microsoft.com/en-us/windows/apps/desktop/modernize/desktop-to-uwp-enhance, since that provides even more details and explanation of platform specific behaviors.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 49205137-dcec-a120-e5e7-1b9c79cdafbb
* Version Independent ID: 4352856b-5ccf-8e62-48b4-3151340cac56
* Content: [Target frameworks in SDK-style projects - .NET](https://docs.microsoft.com/en-us/dotnet/standard/frameworks)
* Content Source: [docs/standard/frameworks.md](https://github.com/dotnet/docs/blob/master/docs/standard/frameworks.md)
* Product: **dotnet**
* Technology: **dotnet-standard**
* GitHub Login: @gewarren
* Microsoft Alias: **gewarren**

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #21304.

### CODE REVIEW

File: docs/standard/frameworks.md [hunk 1]
```diff
@@ -57,13 +58,54 @@ For each .NET 5.0 and later TFM, for example, `net5.0`, there are TFM variations
...
+| net5.0-watchos  | xamarin.watchos (plus everything else inherited from `net5.0`) | |
+| net5.0-windows  | netcoreapp1..3.1 (plus everything else inherited from `net5.0`) | Includes WinForms, WPF, and UWP APIs.<br />For information, see [Call Windows Runtime APIs in desktop apps](/windows/apps/desktop/modernize/desktop-to-uwp-enhance). |
+
+#### Suggested targets
+
+Use these guidelines to determine which TFM to use in your app:
+
+- Apps that are universal or portable to many platforms should target `net5.0`. This includes most libraries but also ASP.NET Core and Entity Framework.
+
+- Platform-specific libraries should target platform-specific flavors. For example, WinForms and WPF controls should target `net5.0-windows`.
```
Comment (adegeo) [inline docs/standard/frameworks.md:83]: "controls" sounds odd here. How about "projects"?
```suggestion
- Platform-specific libraries should target platform-specific flavors. For example, WinForms and WPF projects should target `net5.0-windows`.
```

**Review by adegeo — APPROVED** [review]
Looks good! Left one comment.

File: docs/standard/frameworks.md [hunk 2]
```diff
@@ -57,13 +58,54 @@ For each .NET 5.0 and later TFM, for example, `net5.0`, there are TFM variations
...
+| net5.0-macos    | xamarin.mac (plus everything else inherited from `net5.0`) | |
+| net5.0-tvos     | xamarin.tvos (plus everything else inherited from `net5.0`) | |
+| net5.0-watchos  | xamarin.watchos (plus everything else inherited from `net5.0`) | |
+| net5.0-windows  | netcoreapp1..3.1 (plus everything else inherited from `net5.0`) | Includes WinForms, WPF, and UWP APIs.<br />For information, see [Call Windows Runtime APIs in desktop apps](/windows/apps/desktop/modernize/desktop-to-uwp-enhance). |
+
+#### Suggested targets
+
+Use these guidelines to determine which TFM to use in your app:
+
+- Apps that are universal or portable to many platforms should target `net5.0`. This includes most libraries but also ASP.NET Core and Entity Framework.
```
Comment (vitek-karas) [inline docs/standard/frameworks.md:81]: Nit: I would not use the term "universal" - sounds too much like "UWP". Also so far we've been using "portable" as the term for apps which can run on multiple platforms.

Comment (gewarren) [inline docs/standard/frameworks.md:81]:
```suggestion
- Apps that are portable to multiple platforms should target `net5.0`. This includes most libraries but also ASP.NET Core and Entity Framework.
```

**Pull request merged**